pkgs/os-specific/linux/msi-ec/default.nix

```
@@ -0,0 +1,49 @@
+{
+  stdenv,
+  lib,
+  fetchFromGitHub,
+  linuxPackages,
+  kernel ? linuxPackages.kernel,
+}:
+stdenv.mkDerivation {
+  pname = "msi-ec-kmods";
+  version = "0-unstable-2024-09-19";
+
+  src = fetchFromGitHub {
+    owner = "m1dugh";
+    repo = "msi-ec";
+    rev = "60aad95e2c8948140232f4c4d70f4ceba821d5a5";
+    hash = "sha256-qgM4KeGoLvbVsii5YR3GTGi2MVlU+JeArzEoA+6A3a0=";
+  };
+
+  dontMakeSourcesWritable = false;
+
+  patchPhase = ''
+    cp ${./patches/Makefile} ./Makefile
+  '';
+
+  hardeningDisable = [
+    "pic"
+  ];
+
+  makeFlags = kernel.makeFlags ++ [
+    "KERNELDIR=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
+    "INSTALL_MOD_PATH=$(out)"
+  ];
+
+  nativeBuildInputs = kernel.moduleBuildDependencies;
+
+  buildFlags = [ "modules" ];
+  installTargets = [ "modules_install" ];
+
+  enableParallelBuilding = true;
+
+  meta = with lib; {
+    description = "Kernel modules for MSI Embedded controller";
+    homepage = "https://github.com/m1dugh/msi-ec";
+    license = licenses.gpl2;
+    maintainers = [ maintainers.m1dugh ];
+    platforms = platforms.linux;
+    broken = kernel.kernelOlder "6.2";
+  };
```

```suggestion
  meta = {
    description = "Kernel modules for MSI Embedded controller";
    homepage = "https://github.com/m1dugh/msi-ec";
    license = lib.licenses.gpl2;
    maintainers = [ lib.maintainers.m1dugh ];
    platforms = lib.platforms.linux;
    broken = lib.kernel.kernelOlder "6.2";
  };
```

Nested `with lib;` is discouraged. You might want to be more explicit.